tools/clang_format.rst

```diff
@@ -0,0 +1,44 @@
+.. _using_clang_format:
+
+#######################################
+Using clang-format for LSST Development
+#######################################
+
+The easiest way to comply with the :ref:`LSST code layout rules <style-guide-cpp-6-1>` is to use `clang-format <http://clang.llvm.org/docs/ClangFormat.html>`_.
+
+You can install this tool through the Ubuntu package manager:
+
+.. code-block:: bash
+
+    sudo apt-get install clang-format
+
+or through `Homebrew <http://brew.sh>`_ (on macOS):
+
+.. code-block:: bash
+
+    brew install clang-format
+
+For more platforms see `here <http://releases.llvm.org/download.html>`_.
+
+Place the following configuration file, named ``.clang-format``, in your top-level source directory:
```

Comment:
I am not keen on one `.clang-format` file per repo because keeping them in synch sounds too difficult. Many in-editor tools allow the user to have a single config file, which is my personal preference. For users of `lsstsw` (which is likely a significant fraction of our software developers) we could also provide a .clang-format in the `build` directory.